### DISCUSSION TITLE
Config Menu

### DISCUSSION THREAD

**Discussion by Atari8guy** [frntc/SIDKick-pico#70] [General]
Upon installing this device, I got into the config menu once...since then if I type the SYS 54301 I just get a blank screen and the READY prompt... is that expected?

**Comment by frntc** [comment]
No, of course not :-) How did you install it (additional address lines?)? Can you hear the SID when you start a program (never entering the config menu)?

**Comment by Atari8guy** [comment]
no additional address lines.  I just plugged it in.  the sound works and the mouse works in GEOS, which was my reason for getting it to begin with, so I'm happy and it's working fine as far as I can tell, but I can't get back into the config menu...I downloaded the firmware from here this morning, and as I said, it worked the first time

**Reply by frntc** [reply]
No real idea what went wrong. Have you tried flashing the firmware again? (maybe with slightly lower/higher bus timings, via SKpicopatch)

**Reply by Atari8guy** [reply]
I tried several different timings, i was able to get into the config menu once, but not repeatedly.  The menu was somewhat garbled but I was able to work through it?  I'm also sort of at a loss....I checked all the solder points and reflowed a couple of them, even though they looked okay.  No difference.

**Reply by frntc** [reply]
Usually such problems only occur when the timings are bad (the next firmware will include some minimal diagnostics...). 

If your SKpico-PCB is ok, and other timings did not help... do you have another Pico to test? Maybe writing to the flash doesn't work? (that's what the config menu does)

**Reply by Atari8guy** [reply]
I have another pico w but it has no headers on it.  I suppose I could source another one.  In the mean time I'll give a few more variations on the timing a try - I read through your stuff, you think lower is better?

**Reply by frntc** [reply]
I made a brute force timing diagnostic some time ago, which was inconclusive. But people who ran into timing problems reported that lower values solved their issues.

**Comment by Atari8guy** [comment]
Just to give the last bit of information I can share.  None of the timing changes, up or down, really made a difference in behaviour that I could tell.  I was able after a reflash to get back into the menu one more time, it was garbled again, but I was able to set the SID to 6581, and leave the second one disabled.  I was NOT able to change the timing to NTSC (my machine is NTSC) it stayed on DREAN (I don't even know what that is).  However, once I rebooted the machine I again had no access to the config menu, but outside of that, everything works, SID tunes, demos, and the mouse in GEOS.  I'm not likely to play with it again for a while as aside from the menu access it seems to work and my vacation time is over, I will watch though, in case you do track down the bug or release new firmware.  Thanks for quick responses, even though we didn't get it quite solved.

**Reply by frntc** [reply]
The just updated firmware includes a small diagnostic routine for timings -- you might want to give it a try. Apart from that, it might still be the case that it's a hardware issue (flash etc.).

**Reply by W0RMinator** [reply]
I just wanted to mention that I had the same issue with one of the sidkick pico's that I made. Was not able to change much in the settings and it was stuck on DREAN. Some text was garbled. Like Atari8guy mentioned, sound tests worked fine. I tried reinstalling firmware multiple times and no changes. I've narrowed it down to a hardware issue with the raspberry pi pico board (official board). I desoldered it from the sidkick board and soldered another pi pico board (official) and it worked. Not quite sure what's wrong with the pi pico board but it definitely has some sort of failure. Just posting this in case anyone else has the same issue.
![skpico](https://github.com/user-attachments/assets/fd712ece-3afe-495c-bd0f-6b3a6d05f236)

**Reply by SX64man** [reply]
I have seen the same exact timing issue (thinking it is an Argentinian Drean C64 instead of NTSC) on two of my 5 NTSC machines that have SKPicos with the stereo setup. My several PAL machines appear to be more forgiving and have no issues with the default 15,11 f/w timing settings. However, there appear to be subtle variations of the combined hardware which create a borderline timing condition on some NTSC machines. I have tried to pick up these subtle variations on my scope but it has been inconclusive so far.  All SK boards were made from scratch with identical components. The only workaround I have found is to reflash f/w with various combinations of 15/11/100, 13/9/100, 11/7/100, 10/6/100 in the batch file until one appears to work without channel clicks or dropouts, menu corruption, or error message.  The 100 setting is to pass the diag test (a setting of 20 offset did not work for me with my test harness).  The 100 appears to work on several machines so far and I plan to flash the other 5 to see if all the diag runs pass.  All my boards are h/w rev 0.1 with FW0.211   
FRNTC, I put the SKpicotiming.prg util as one of the PRG entries in the F7 menu and that comes in handy.     It would be awesome if this utility had the ability to read the current value in the firmware, display it, and then give you the option to write new values, if possible.  It would make it easier to test and document the best timing values for a specific C64.  Thanks for the great work!

**Reply by MarkMccabe** [reply]
I have the same issues on some of the ones I built and tested in my NTSC 326298 5 pin video mainboard. It comes up with the menu garbled, thinks it a Drean machine (which is a PAL system), and it will not save changes made. Funny thing is it functions fine, plays music and passes the diagnostic cartridge test (both the SID, sound,  and control port checks for the paddles) perfectly.
It's funny because on the ones I built where the menu comes up fine (sometimes after some real time timing tweaks, I hav
e yet to get the paddles tp pass the diagnostic tests... Confusing...

**Reply by frntc** [reply]
The Drean vs. NTSC detection might be an issue of the VIC-type-detection routine. As I don't have a 326298-NTSC nor a Drean machine the routine might have a bug (which I either did not test in Vice, or didn't pop up there).

The garbled/non-starting menu is a matter of bus timing values.

**Comment by banman11** [comment]
Very interesting....
I must admit I am having the exact same issue as Atari8guy. Only one of my 5 boards showed the configuration menu. The other 4 would refresh the screen with the 'ready' prompt and flashing cursor.

I thought I'd ad to this conversation by enlisting the help of AI.
it might be of some insight to other people who have similar issues.

https://gemini.google.com/share/39a7371a9daf

**Reply by frntc** [reply]
I can't tell much without knowing which variant of the SKpico you have (integrated, 2040/2350, adapter board: which Pico(-clone), ...)...?

But what I can say is that ~70% of the AI-answer is completely and factually incorrect ;-)

**Reply by banman11** [reply]
Hi frntc,

Yes that AI can output some unusual information.

Thanks for the heads up on this AI information


I will check which sidekick board I have. I am fairly certain that it's the earlier variants that use the Raspberry Pi Pico which just plugs into the Sidkick board.
I am using cheap clone Raspberry Pi Pico's (the black coloured variety). The level shifter IC's I'm using are cheap AliExpress varieties as well.

At a guess it's either the cheap Pico boards and/or the cheap AliExpress level shifter IC's that are the issue. 
I actually have ordered official Raspberry Pi Pico' boards from a registered reseller here in Oz. Hopefully they arrive soon and I can try these ones.
I also need to make sure I have the latest firmware version installed on the Raspberry Pi Pico boards

**Reply by frntc** [reply]
Yes, it makes sense to check all of this!

**Reply by banman11** [reply]
Hi frntc,

I have a very interesting update.
I am going to take back all my assumptions unreservedly.
I might add this a cautionary tale as well......

Today I got my official Raspberry Pi Pico board (I also got a pico2 and a Pico probe) from Core electronics here in Oz.
I programmed the official board with your latest firmware version. I plugged it in and it worked fine, 100% . I tried a few different C64 machines.

The menu comes up no problems. Now I did nothing to any of the settings, just the default settings.
I tried 2 boards no problems. I have a 3rd one that I used turned pin header sockets but that had connector issues. I would only suggest using these machined turned pin sockets if one was going to be using them a maximum of about 5 insertions.
In my opinion stick with the double wipe sockets.
I have cut these off as you can see in the pictures and I will be using double wipe sockets.

With the fourth Sidkick board I was so obsessed with my soldering and level shifter IC's that I removed them at great pains. So I need to carefully put them back and test. I would, on the balance of probabilities, say it will work as well.


Now, that the problem which really wasn't a problem has been solved. I have a few questions.....

I did have a play around with the timing.prg, very neat and interesting.

Evidently, the clone, knockoff, fake (whatever you like to call them) Pi Pico boards were at fault. However they all played the music fine and the RGB led's did their thing. I just was not able to access the configuration menu. 
I must have here with me about 6 of these Pi Pico boards. Different types. Ones that look like the original Pi Pico (green boards with 512kb memory, I think), the black RGB led 4mb versions.
I still need to test all of them. Of the 4 I used so far only one (black board 4mb RGB led) seemed to be fully functional. That is to say it was able to display the configuration menu.

What could be going on here with these boards? Is it like what the AI client was talking about, that is slower flash memory on these clones.
Is there a way to get these things to display the configuration menu.

Any thoughts would be interesting to chase up.

**Comment by banman11** [comment]
![IMG_20260410_150620-750x1000](https://github.com/user-attachments/assets/402bf5e8-d96c-4d6d-8568-84527d0471e0)
![Uploading IMG_20260410_150623-750x1000.jpg…]()

**Reply by frntc** [reply]
Yes, some clones use slower flash ICs, but the config tool/communication does not happen from flash but from RAM (the slow flash can cause boot problems, but these are fixed since some firmware-version quite a while ago). The clones seemingly are of low production quality, maybe it's also the routing on the PCBs. Seems that I was mostly lucky with those.

Playing and LED _seemingly_ work also when there's a small problem, as some wrong signals often don't cause immediately audible/visible problems. But a single flipped bit can results in a misfunctional config tool in the C64's RAM.

**Comment by banman11** [comment]
![Uploading IMG_20260410_150625-750x1000.jp
![IMG_20260410_150628-750x1000](https://github.com/user-attachments/assets/8503c7a8-564b-4e7b-a54d-0d54c4625c48)
g…]
![IMG_20260410_150700-750x1000](https://github.com/user-attachments/assets/7d4de5b4-9968-4e7e-be27-92fdfad37670)
()

**Reply by banman11** [reply]
That's very interesting frntc. Thanks for the feedback.
Yes that make sense. As the clone Pi Pico boards all seem to play music and the LED's seem to function as expected. Just trouble accessing the configuration menu.
To date it was just that one clone Pi Pico board that worked correctly....

I have a few more clone boards that I haven't tried yet. I'll go thru them one by one and observe them. 
So essentially (in my case at least) it comes down to faulty clone Pi Pico boards.
Potentially, if my sampling is weighted towards clone Pi Pico boards that are failing more than less, this could be what's giving other users these issues. That is to say the sounds play fine but no access to the configuration menu.

Is there a way to get the firmware to compensate for these clone Pi Pico boards or are they just going to be partially operational?

Okay I'll get back to you on my sampling....

**Comment by banman11** [comment]
![Uploading IMG_20260411_161247.jpg…]()
![IMG_20260411_161247](https://github.com/user-attachments/assets/aa10e4c2-31d9-431b-b9f3-9d2031f89f38)

Okay I think I have collected all the clone Raspberry Pi Pico boards I could find.
There's 4 x RGB led 4mb clone Raspberry Pi Pico (black) and 4 X clone regular style (512mb green varieties I think).
It might take some time to put them together. I need to order some more pin headers.

**Reply by banman11** [reply]
I have started to test several on these clone Pi Pico boards.
A Lille but of background information about these.
I purchased these all from AliExpress. They were the cheapest price, purchased randomly from different vendors at different times.
Hope this will be a good blind sample set.

**Comment by rittwage** [comment]
Is this really neccasary? I mean it's just FAFO if you want to build using a clone board.

**Reply by banman11** [reply]
That’s a fair point, Jim. I suppose that’s the classic 'FAFO' tax—saving a few dollars on a clone board only to pay for it later in troubleshooting time! It definitely adds an annoying wildcard when you’re trying to dial in these bus timings. I’ll stick to the official Pico for the next build to keep the variables to a minimum. Cheers!

**Reply by MarkMccabe** [reply]
I have built 10 of these guys with the Black clone. 7 worked fine, 3 had the dreaded "Drean" issue. Friends that have built theirs with the "official" Raspberry Pi Pico have also seen the dreaded "Drean" issue. All were tested in NTSC machines...

**Reply by banman11** [reply]
[](url)That's very interesting....

Actually I'm just thinking out loud here. 
I had a totally unrelated project where I needed to program an RP2350 via SWD. I thought I would repurpose a previously used one (Pi Pico) from a Sidkick into a Picoprobe.
I had a terrible time trying to get it to connect.
I tried everything, it would behave normally as far as the PC was concerned. That is to say the device showed up as a correct cmsis DAP probe under windows. It would accept all the commands. It just would not see or connect to the target device.
That's what got me into this thread. I started realising that there was something wrong with the Raspberry Pi Pico boards and not my shambolic Sidkick construction skills. 
Somewhere in the back of my mind something was biting at me.
Here's a thought, potentially the SidKick board could detect issues with Raspberry Pi Pico's that were not quite right. Whether they are factory original or clones/ knockoffs.

**Comment by rittwage** [comment]
Is it possible to just run the config tool as a regular loaded program and not rely on the SYS behavior?

**Reply by banman11** [reply]
Okay that's interesting...
Would the config tool be located in the current firmware release. That is to say, is it SKpicoconfig.prg?
 I'd like to give it a try...

**Comment by banman11** [comment]
I have tested 5 clone Pi Pico boards so far. I am now waiting on parts for the remaining 3..

It is  a mixed bag.

https://imgur.com/a/HjIxijx

**Reply by banman11** [reply]
I was having some issues posting images so I reverted to Imgur to share my images.

At this point 2 of the green clone Pi Pico boards (512kb) appear to be okay.

1 black clone Pi Pico board (2mb variety ?) Will play sound okay. However it sometimes may not go into the configuration menu. Let's say 50% of the time.

1 black clone Pi Pico board plays sound but will refuse to got into the configuration menu.

1 green clone Pi Pico board will not work at all. It will cause the C64 to black screen. Very occasionally it will display a locked up boot screen with what I would describe as a flipped bit (faulty bit) on the character ROM display.

The 2 clones I was using to program a RP2350  via SWD happened to be the clone green Pi Pico board that won't boot the C64 and the clone black Pi Pico board that has sound but won't show the configuration menu..

**Reply by banman11** [reply]
This Pi Pico board looks like a  bent raw prawn... Hahahaha 😂 🤣😂🤣😂.

However it seems work...

https://imgur.com/a/p1c9xqg

**Comment by banman11** [comment]
A small update....

I tested what Sidkick thought was a bad clone Pi Pico AND a good clone Pi Pico on a One ROM Ice device using One ROM Studio. I re flashed the Picoprobe firmware to both of these clone Pi Pico boards successfully.

The dodgy Pi Pico board would not communicate with the One ROM Ice device via SWD even though the Pi Picoprobe was registered to windows 10 correctly and One ROM Studio could communicate okay using USB cable.

The good Picoprobe (as determined by Sidkick) did successfully communicate with the One ROM Ice device thru One ROM Studio via SWD with out issue.
I have posted photos here if interested....

https://imgur.com/a/mMewV8U

**Reply by banman11** [reply]
I had actually thought that there was something wrong with my One ROM setup previously. I had been at it for months unsuccessfully. 
Using the Sidkick technique it probably took me a lazy 30 mins to determine which clone Pi Pico boards were acceptable and which weren't (that is to say which Pi Pico's were faulty and which were good).

**Reply by banman11** [reply]
It’s a classic case of "penny wise, pound foolish" when it comes to those clone boards. It’s actually quite impressive that the SIDKick pushes the hardware so hard that it reveals manufacturing tolerances that even a standard microcontroller project would never notice.
Since the boards are already built and I believe I know the issue is the flash latency/quality on the clones, I might actually be able to repurpose them for other projects that aren't as "timing-aggressive."

**Reply by frntc** [reply]
I'm lacking the time currently to follow this thread, but I need to quickly "intervene".

The flash is only read during startup, and non-time critically when reading the PRGs/config tool to be launched. Everything else runs from RAM only. I don't expect any problems with reading the flash, if so, it would make sense to document the exact flash IC designations.

I can't say much about your selection of clones (got lost here), but I would not say that the SKpico pushes hardware really hard. Many other projects push much harder. 

I have 10-20 of those black clones and the only thing I encountered was the boot sequence thing related to the flash (which is fixed in the firmware). In a forum there was a collective purchase order with >100 SKpicos with the black clones, and no issues (again, except the one which has been fixed). This does not mean, I'd say, that poor production quality can now lead to more problems. In any case, I'd be happy if people move from the large, older SKpicos to the new integrated version... makes life easier for me to maintain fewer variants...

About the "Drean"-issue: The SKpico does not detect whether it's a PAL, DREAN or NTSC system. The config tool, which is a plain PRG running on the C64, does that. If launching the config tool does not work properly, then the problem is there.

**Reply by banman11** [reply]
Hi frntc,
Point taken regarding the flash; we can discount that as a variable. I totally understand how busy you must be, and I really appreciate you taking the time to weigh in.
My main interest in these observations is the SIDKick’s apparent ability to act as a "suitability tester" for these clone/knockoff Pi Pico boards. It seems your project pushes the RP2040 and its bus timing to a threshold that standard "blink" tests or even basic emulation just don't reach.
It’s a fascinating side effect—the SIDKick basically becomes a high-stress benchmark for signal integrity on these boards. Have you noticed other users reporting this "binary" success rate with clones? I'm curious if you think this comes down to the silicon quality of the RP2040 clones themselves, or if the PCB routing on these cheaper boards is simply failing at the frequencies you're hitting
